File: solutions/python/wordy/1/wordy.py
def answer(question):
    operation = ["plus","minus","multiplied","divided"]
    words = question.replace("?","").replace("by","").split()
    if words[0:2] != ["What","is"] or "cubed" in words:
        raise ValueError("unknown operation")
    if len(words)<3 or words[2].isalpha():
        raise ValueError("syntax error")
    res = int(words[2])
    words = words[3:]
    while len(words)>0:
        try:
            if words[0] == "plus":
                res += int(words[1])
            elif words[0] == "minus":
                res -= int(words[1])
            elif words[0] == "multiplied":
                res *= int(words[1])
            elif words[0] == "divided":
                res //= int(words[1])
            else:
                raise ValueError("syntax error")
            words = words[2:]
        except:
            raise ValueError("syntax error")
    return res

File: solutions/javascript/wordy/1/wordy.js
export const answer = (operation) => {
  if (!operation.startsWith("What is") || !operation.endsWith("?")) {
    throw new Error("Unknown operation");
  }
  const opArray = operation
    .replace("?", "")
    .replace("What is ", "")
    .replaceAll("by ", "")
    .split(" ");

  if (opArray.length === 0) throw new Error("Syntax error");

  let acc = Number(opArray[0]);

  for (let i = 1; i < opArray.length; i++) {
    if (!isNaN(opArray[i]) && !isNaN(opArray[i - 1]))
      throw new Error("Syntax error");
    if (isNaN(opArray[i]) && isNaN(opArray[i - 1]))
      throw new Error("Syntax error");

    if (isNaN(opArray[i])) {
      if (!["plus", "minus", "multiplied", "divided"].includes(opArray[1])) {
        throw new Error("Unknown operation");
      }

      if (i + 1 >= opArray.length || isNaN(opArray[i + 1])) {
        throw new Error("Syntax error");
      }

      if (opArray[i] === "plus") acc += Number(opArray[i + 1]);
      else if (opArray[i] === "minus") acc -= Number(opArray[i + 1]);
      else if (opArray[i] === "multiplied") acc *= Number(opArray[i + 1]);
      else if (opArray[i] === "divided") acc /= Number(opArray[i + 1]);
    }
  }

  return acc;
};
